File: examples/nodejs/src/snippets/document/document-classifier.ts
import * as ScanbotSDK from "scanbotsdk";

export class DocumentClassifierSnippet {
  public static async run(image: ScanbotSDK.ImageRef): Promise<void> {
    const licenseInfo = await ScanbotSDK.getLicenseInfo();
    if (licenseInfo.status !== "OKAY") {
      console.warn("License is not valid.");
      return;
    }

    ScanbotSDK.autorelease(async () => {
      var config = new ScanbotSDK.DocumentClassifierConfiguration();

      // `await using` ensures both classifier and result are properly disposed
      // when the scope ends, as they hold unmanaged resources.
      await using classifier = await ScanbotSDK.DocumentClassifier.create(config);
      await using result = await classifier.run(image);

      console.log("Detection status: " + result.status);
      console.log("Type: " + result.documentType);

      result.documentScanningResult.detectionResult.points.forEach((p) => {
        console.log(`x: ${p.x}, y: ${p.y}`);
      });
    });
  }
}

File: examples/python/snippets/document/document_classifier.py
from scanbotsdk import *

def classify_document(image: ImageRef):
    config = DocumentClassifierConfiguration(
        crop=True
    )
    
    classifier = DocumentClassifier(configuration=config)
    result: DocumentClassifierResult = classifier.run(image=image)
    
    print(f"Detection status: {result.status.name}")
    print(f"Type: {result.document_type.name}")
    
    if result.document_scanning_result:
        print("Detected points:")
        for point in result.document_scanning_result.detection_result.points:
            print(f"\tx: {point.x}, y: {point.y}")
    else:
        print("No document corners detected.")

